File: qwencoder-eval/chat/eval-dev-quality/testdata/golang/light/largestProperDivisor.go
package light

import "fmt"

func largestProperDivisor(aNumber int) (int, error) {
	if aNumber < 1 {
		return 0, fmt.Errorf("Argument must be >= 1: %d", aNumber)
	}

	if (aNumber & 1) == 0 {
		return aNumber >> 1, nil
	}

	for p := 3; p*p <= aNumber; p += 2 {
		if aNumber%p == 0 {
			return aNumber / p, nil
		}
	}

	return 1, nil
}

File: qwencoder-eval/instruct/eval-dev-quality/testdata/java/light/src/main/java/com/eval/LargestProperDivisor.java
// Copyright (C) rosettacode.org
package com.eval;

public class LargestProperDivisor {
  static int largestProperDivisor(int aNumber) {
    if (aNumber < 1) {
      throw new IllegalArgumentException("Argument must be >= 1: " + aNumber);
    }

    if ((aNumber & 1) == 0) {
      return aNumber >> 1;
    }
    for (int p = 3; p * p <= aNumber; p += 2) {
      if (aNumber % p == 0) {
        return aNumber / p;
      }
    }
    return 1;
  }
}
